### PR TITLE
Adding WASM support by using containerd-wasm-shim

### PR DESCRIPTION
**Pull request by fcabrera23**
- Experimental feature
- Support only for Linux nodes
- containerd-wasm-shim is in alpha version

### CODE REVIEW

**Review by parameshbabu — APPROVED** [review]

File: samples/wasm/Set-AksEdgeWasmRuntime.ps1 [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+    Write-Host "    K8s version found" -ForegroundColor Cyan
+    $IsK8s = $true
+}
+elseif ($version.Contains("K3s"))
+{ 
+    Write-Host "    K3s version found" -ForegroundColor Cyan
+}
+else
+{
+    throw $("AKS Edge Essentials verison not supported")
```
[the remark this answers is not in the document]
Comment (fcabrera23) [reply samples/wasm/Set-AksEdgeWasmRuntime.ps1:51]: Changed

File: samples/wasm/Set-AksEdgeWasmRuntime.ps1 [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+param(
+    [Switch] $enable,
+    [string] $shimVersion = "v0.3.3",
+    [ValidateSet("spin", "slight", "both")]
+    [string] $shimOption = "both"
+)
+
+Write-Host "1. Checking AKS Edge Essentials dependencies" -ForegroundColor Green
+$IsK8s = $false
+$version = (Get-ChildItem -Path 'HKLM:\SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\' | Get-ItemProperty |  Where-Object {$_.DisplayName -like 'AKS Edge Essentials*'}).DisplayName
```
[the remark this answers is not in the document]
Comment (fcabrera23) [reply samples/wasm/Set-AksEdgeWasmRuntime.ps1:33]: Changed

File: samples/wasm/README.md [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+
+## Instructions
+
+1. Setup AKS Edge Essentials - Follow [Set up machine](https://aka.ms/aks-edge/quickstart)
+1. Deploy a [Single Machine cluster](https://learn.microsoft.com/azure/aks/hybrid/aks-edge-howto-single-node-deployment) or a [Scalable Cluster](https://learn.microsoft.com/azure/aks/hybrid/aks-edge-howto-multi-node-deployment).
+1. Open an elevated PowerShell session
+1. Move to an appropriate working directory
+1. Download [Set-AksEdgeWasmRuntime.ps1](./Set-AksEdgeWasmRuntimes.ps1)
+    ```powershell
+    Invoke-WebRequest -Uri "https://raw.githubusercontent.com/Azure/AKS-Edge/wasm-enablement/samples/wasm/Set-AksEdgeWasmRuntimes.ps1" -OutFile ".\Set-AksEdgeWasmRuntimes.ps1"
```
[the remark this answers is not in the document]
Comment (fcabrera23) [reply samples/wasm/README.md:22]: Added

File: samples/wasm/README.md [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+6. Deploy Wasm workloads to your cluster using the *wasmtime-spin* and *wasmtime-slight* runtime classes deployed in the previous step.
+
+    ```powershell
+    kubectl apply -f https://raw.githubusercontent.com/Azure/AKS-Edge/wasm-enablement/samples/wasm/workload.yaml
+    ```
+
+7. Check that the pods are deployed and running
+
+    ```powershell
+    kubectl get pods
```
[the remark this answers is not in the document]
Comment (fcabrera23) [reply samples/wasm/README.md:60]: Yes - Changed yaml files to deploy under "wasm" namespace and updated instructions

File: samples/wasm/README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (fcabrera23) [reply samples/wasm/README.md:22]: Changed wasm-enablement to "preview" - Editing PR from main to preview

File: samples/wasm/Set-AksEdgeWasmRuntime.ps1 [hunk 5]
```diff
@@ -0,0 +1,200 @@
...
+    Write-Host "    K3s version found" -ForegroundColor Cyan
+}
+else
+{
+    throw $("AKS Edge Essentials verison not supported")
+}
+
+if ($enable.IsPresent)
+{
+    Write-Host "2. Downloading shim verison $shimVersion" -ForegroundColor green
```
[the remark this answers is not in the document]
Comment (fcabrera23) [reply samples/wasm/Set-AksEdgeWasmRuntime.ps1:56]: Yes - I want to get some customers to try it out and see if they get errors based on their deployments. Once we have feedback on potential errors, I'll move all into one script that we copy into the machine, run it, and get a status code.

**Pull request merged**